src/action.c

```diff
@@ -746,6 +748,33 @@ actions_run(struct view *activator, struct server *server,
 				view_move_relative(view, x, y);
 			}
 			break;
+		case ACTION_TYPE_MOVETO_CURSOR:
+			if (!view) {
+				break;
+			}
+			int x = server->seat.cursor->x - (view->pending.width / 2);
+			int y = server->seat.cursor->y - (view->pending.height / 2);
+			struct output *pending_output = output_nearest_to_cursor(server);
+			if (x + view->pending.width
+					> pending_output->usable_area.x
+					+ pending_output->usable_area.width) {
+				x = pending_output->usable_area.x
+					+ pending_output->usable_area.width - view->pending.width;
+			}
+			if (x < 0) {
+				x = 0;
+			}
```

Comment:
This could be `x = MAX(x, 0)`, same for the `y` case below.
Both cases should also use the x/y coordinates of the usable area instead of `0` (in case there is some panel docked). E.g.

`x = MAX(x, pending->output->usable_area.x);` in this case.